### PR TITLE
feat: add Databricks ODBC TPCDS benchmark

### PR DESCRIPTION
**Pull request by peasee**
# 🗣 Description

* Adds the TPCDS benchmark for ODBC Databricks.

## 🔨 Related Issues

* Part of #3351

### CODE REVIEW

File: crates/runtime/benches/utils/mod.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ pub(crate) fn init_tracing(trace_config: Option<&str>) {
         (_, Some(log)) => EnvFilter::new(log),
         (Some(level), None) => EnvFilter::new(level),
         _ => EnvFilter::new(
-            "runtime=TRACE,datafusion-federation=TRACE,datafusion-federation-sql=TRACE,bench=TRACE,INFO",
+            "runtime=TRACE,datafusion-federation=TRACE,datafusion-federation-sql=TRACE,bench=TRACE",
```
Comment (phillipleblanc) [inline crates/runtime/benches/utils/mod.rs]: Why was this removed?

**Review by phillipleblanc — DISMISSED** [review]

File: crates/runtime/benches/utils/mod.rs (hunk 1 above)
Comment (peasee) [reply crates/runtime/benches/utils/mod.rs]: `INFO` was enabling `task_history` log spans, which was causing a lot of logs in the benchmark runs making it hard to debug. I've updated with a new log span copying the default from `spiced` but disabling `task_history` - I'm testing the output of this new span now to see what it looks like

**Review by phillipleblanc — APPROVED** [review]
